modules/storage/local.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package storage
+
+import (
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+)
+
+var (
+	_ ObjectStorage = &LocalStorage{}
+)
+
+// LocalStorage represents a local files storage
+type LocalStorage struct {
+	dir string
+}
+
+// NewLocalStorage returns a local files
+func NewLocalStorage(bucket string) (*LocalStorage, error) {
+	if err := os.MkdirAll(bucket, os.ModePerm); err != nil {
+		return nil, err
+	}
+
+	return &LocalStorage{
+		dir: bucket,
+	}, nil
+}
+
+// Open a file
+func (l *LocalStorage) Open(path string) (io.ReadCloser, error) {
+	f, err := os.Open(filepath.Join(l.dir, path))
+	if err != nil {
+		return nil, err
+	}
+	return f, nil
+}
+
+// Save a file
+func (l *LocalStorage) Save(path string, r io.Reader) (int64, error) {
+	p := filepath.Join(l.dir, path)
+	if err := os.MkdirAll(filepath.Dir(p), os.ModePerm); err != nil {
+		return 0, err
+	}
+
+	// always override
+	if err := os.RemoveAll(p); err != nil {
```

Comment:
Given my recent PR you probably need to use modules/util.RemoveAll here

```suggestion
	if err := util.RemoveAll(p); err != nil {
```